scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,288 @@
+                        pet:castSpell(level >= 80 and 6 or level >= 61 and 5 or level >= 41 and 4 or level >= 21 and 3 or level >= 11 and 2 or 1, member)
+                        return true
+                    end
+                end      
+            end                       
+            for _,member in ipairs(party) do                
+                if math.random(0,99) < 50 then -- pick a player somewhat at random                    
+                    casted = buffPlayer(member,pet,level)
+                    if casted == true then return true end
+                end
+            end
+            if masterHP2 then -- low cure                
+                pet:castSpell(level >= 30 and 3 or level >= 20 and 2 or 1, master)
+                return true
+            end                       
+        end        
+    end
+    return false
+end;
+
+function buffPlayer(player,pet,level)        
+    if not player:hasStatusEffect(dsp.effect.PROTECT) and level >= 7 then -- protect        
+        local sp = level >= 76 and 47 or level >= 63 and 46 or level >= 47 and 45 or level >= 27 and 44 or 43                
+        pet:castSpell(sp, player)
+        return true
+    end
+    if not player:hasStatusEffect(dsp.effect.SHELL) and level >= 17 then -- shell        
+        pet:castSpell(level >= 76 and 52 or level >= 68 and 51 or level >= 57 and 50 or level >= 37 and 49 or 48, player)
+        return true
+    end
+    if not player:hasStatusEffect(dsp.effect.HASTE) and level >= 40 then -- haste        
+        pet:castSpell(57, player)
+        return true
+    end
+    if not player:hasStatusEffect(dsp.effect.REGEN) and level >= 21 then -- regen        
+        pet:castSpell(level >= 86 and 477 or level >= 66 and 111 or level >= 44 and 110 or 108, player)
+        return true
+    end
+    return false
+end;
+
+function getGearMod(master)
+    -- Summoner's spats
+    local leg = master:getEquipID(dsp.slot.LEGS);
+    if leg == 15131 or leg == 15594 then
+        return -5000
+    end    
+    return 0
+end;
+
+function getSummoningSkillOverCap(avatar)
+    local summoner = avatar:getMaster()
+    local summoningSkill = summoner:getSkillLevel(dsp.skill.SUMMONING_MAGIC)
+    local maxSkill = summoner:getMaxSkillLevel(avatar:getMainLvl(), dsp.job.SMN, dsp.skill.SUMMONING_MAGIC)
+    return math.max(summoningSkill - maxSkill, 0)
+end;
+
+function getTimeCost(avatar)
+    local summoner = avatar:getMaster()
+    local summoningSkill = summoner:getSkillLevel(dsp.skill.SUMMONING_MAGIC)
+    local maxSkill = summoner:getMaxSkillLevel(avatar:getMainLvl(), dsp.job.SMN, dsp.skill.SUMMONING_MAGIC)
+    -- 45 s +/- 1 second for every 3 skill over or under cap
+    return 45000 - ((summoningSkill - maxSkill)/3) * 1000
+end;
+
+function getElement(avatar)
+    --avatar:getPetElement()     this doesn't return anything but zer0
+    if (avatar:getFamily() == 99) then             
+        return dsp.subEffect.WIND_DAMAGE
```

Comment:
Yeah, after having wrote this I think this is a good modification, I was trying to stick with stuff that already exists, as my lua skills are .07a.

Reply:
It comes for free from `magic.lua`:
https://github.com/DarkstarProject/darkstar/blob/master/scripts/globals/magic.lua#L16

Reply:
https://github.com/DarkstarProject/darkstar/blob/master/scripts/globals/status.lua#L106
dsp.subEffect.LIGHTNING_DAMAGE = 5

https://github.com/DarkstarProject/darkstar/blob/master/scripts/globals/magic.lua#L24
dsp.magic.element.LIGHTNING = 6

https://github.com/DarkstarProject/darkstar/blob/master/sql/mob_family_system.sql#L170
.....,5,32,0)
5 being Lightning's DB element for mob family, that's what I was going based on.  Assuming the mobentity::element property was getting set (which lua script wasn't pulling it)

Reply:
I personally feel that the "element" of a spirit _should_ be _thought of_ as "magical element" rather than "the animation ID associated with that element". However, mob family does seem to be using 5 to represent the element. I sense a can of worms, so I guess just leave it as is for now and wait for _someone else_ to say "time to consolidate this stuff so it matches".